storage/swift_storage.py

```diff
@@ -171,8 +178,11 @@ def get_sanitized_uri(self) -> str:
     def _find_storage_objects_with_prefix(
             self, container: str, prefix: str) -> List[Dict[str, str]]:
         connection = self.get_connection()
-        _, container_objects = connection.get_container(container, prefix=prefix)
-        return container_objects
+        try:
+            _, container_objects = connection.get_container(container, prefix=prefix)
+            return container_objects
+        except ClientException:
+            raise NotFoundError("No Files Found")
 
     def save_to_directory(self, directory_path: str) -> None:
```

Comment:
I can't comment on line 192, but should we raise if `_find_storage_objects_with_prefix` returns an empty list (or iterator)?